### DISCUSSION TITLE
Add context code only argument to instructions

### DISCUSSION THREAD

**Discussion by arthuro555** [4ian/GDevelop#3193] [Archive] [answered]
## Description
While making extensions, I sometimes wish instructions would have a **context**. This would make things like a trigger once easier to make and not require a custom code generator. Plus I think it isn't possible to use a custom code generator in Js (setCustomCodeGenerator not exposed).

## Solution suggested
Adding a context parameter. The context would be a reference to an object passed through a code only parameter. This object is specific to a call of the instruction. You could then store a variable like a uid in it to separate multiple calls for example. This would work easily by making the code generator define a new unique object when such a parameter is given and pass that object as argument.

## Alternatives considered
Instead of a context object a uid object would be fine too, but not as hands as a context object.

**Comment by Bouh** [comment]
We should keep the issue tab on github for the issues.
For echange about a new features or options or just suggestions we have the trello or the forum.
In general a new feature is added when multiples users need it.

**Comment by arthuro555** [comment]
I thought that as this is a "developer" feature request it would be more appropriate on github. Thanks for the clarification.

**Comment by 4ian** [comment, marked as answer]
Ideally GitHub would add in the future a "Discussion" tab (I think they might have been working on this). For now, not sure what is the better between "polluting" issues or making technical discussion on the forum. What do you think?

**Comment by arthuro555** [comment]
Maybe there could be a "Contributors" group on the forum and a new "Technical Forum" reserved to that group. Any new contributor can ask for getting into that group (it not being public prevents from having missorted topics on the forum (Non-devs posting a non technical discussion), but of course it can also be public.)

**Comment by arthuro555** [comment]
@4ian about the discussion tab, it is already present for teams in an organization. It will probably come out this year for user repos too. It would be neat to either have a GitHub team/organization for discussion of that kind of stuff or something similar to what I proposed above.
